client/src/Components/ShippingInformation.jsx

```diff
@@ -0,0 +1,186 @@
+
+class ShippingInformation extends React.Component{
+  constructor(props){
+    super(props)
+    this.state = {
+      shippingTime: [],
+      exclusions: [],
+      returnPolicy : [],
+      returnOverview : [],
+      giftWrapping: [],
+      getShippingCost: false,
+      shippingCost: [],
+      enteredShippingInfo: false,
+      zipCode : [],
+      shippingFrom: [],
+      shippingTo: [],
+      shippingToCountry: [],
+      deliveryDate : []
+    }
+    this.getShippingInformation = this.getShippingInformation.bind(this);
+    this.getReturnPolicy = this.getReturnPolicy.bind(this);
+    this.getShippingInfo = this.getShippingInfo.bind(this);
+    this.showShipping = this.showShipping.bind(this);
+    this.shippingCost = this.shippingCost.bind(this);
+    this.shippingToCountry = this.shippingToCountry.bind(this);    
+  }
+
+
+  componentDidMount(){
+    this.getShippingInformation()
+    this.getReturnPolicy()      
+  }
+  getShippingInformation(){
+    let manufacturingTime;
+    let shippingFrom = this.props.data[0]['shippingFrom']
+    let shippingTo = this.props.data[0]['shippingTo']
+    let deliveryDate = this.props.data[0]['deliveryDate']
+    manufacturingTime = this.props.data[0]['manufacturingTime'];
+    this.setState({
+      shippingTime: manufacturingTime,
+      shippingFrom: shippingFrom,
+      shippingTo: shippingTo,
+      deliveryDate : deliveryDate
+    })
+  }
+
+  getReturnPolicy(){
+    var returnable;
+    var returnPolicy ="";
+    var exclusion;
+    var wrapping = "";
+    if (this.props.data[0]['availableToReturn']){
+      returnable = "Returns and exchanges accepted"
+      exclusion = "Exceptions may apply. "
+      returnPolicy ="See return policy"
+      wrapping = "Gift wrapping available"
+    } else{
+      returnable ="No returns or exchanges"
+      exclusion = "But please contact me if you have any problems with your order."
+    }
+    this.setState({
+      exclusions : exclusion,
+      returnPolicy : returnable,
+      returnOverview : returnPolicy,
+      giftWrapping : wrapping
+    })
+  }
+
+  getShippingInfo(){
+    event.preventDefault();
+    this.setState({
+      getShippingCost: !this.state.getShippingCost
+    })
+  }
+
+  showShipping(){
+    event.preventDefault();
+  }
+  
+  shippingCost(){
+    let shippingCost = this.props.data[0]['shippingCosts'];
+    if (typeof parseInt(event.target.value) === 'number' && event.target.value.length === 5){
+      this.setState({
+        shippingCost: shippingCost,
+        enteredShippingInfo: true,
+        getShippingCost: !this.state.getShippingCost,
+        zipCode: event.target.value
+      })
+    }
+  }
+
+  shippingToCountry (){
+    this.setState({
+      shippingToCountry: event.target.value
+    })
+  }
+
+  render(){
+    return(
+    <div>
+    <div className="shippingInformation">
+      <div style={{fontWeight: 'bold'}}>
+        Shipping & Returns
+      </div>
+      {!this.state.enteredShippingInfo ?
```

Comment:
this is a *really* hard to read ternary statement.  I recommend you switch it to an if/else conditional